### DISCUSSION TITLE
Migrating Synology Photos -> Photoprism

### DISCUSSION THREAD

**Discussion by Vlayke** [photoprism/photoprism#1974] [Q&A] [answered]
Hi,

I've been reading about Photoprism a lot lately and I'm considering it as a replacement for Synology Photos that I'm using currently. The one thing that's holding me back is uncertainty about how the migration process would look. Is there even such a thing, really?
In Synology Photos I invested a lot of time into face-recognition organization. Tagging the faces and so forth. To make things "worse" so have my wife and mother-in-law who are two other heavy-users of Photos.
I would definitely want to avoid having them to re-do all this work. So I'd like to know if it's possible to migrate facial recognition data or is that a no-go?
Also what would happen with things like albums and so on?

Thanks,
Vlayke

**Comment by graciousgrey** [comment, marked as answer]
Hi,

PhotoPrism is reading a lot of existing metadata from different formats such as exif, iptc or json.
You can find a list of all supported metadata fields in our docs: https://docs.photoprism.app/user-guide/advanced/metadata/.

A lot of additions regarding reading in faces are planned as well: https://github.com/photoprism/photoprism/issues?q=is%3Aissue+is%3Aopen+people+import+in%3Atitle+

If this is possible for Synology depends on how Synology saves the information you add using the tool. Is it saved in their database only? Do they write the metadata to exif? Is there an option to export metadata?

**Reply by Vlayke** [reply]
As far as I'm aware (I could be wrong) Synology Photos (as well as previous applications, like Photo Station and Moments) don't touch the files directly, but rather store data in the Synology internal Postgres DB.

I don't have the knowledge to mess around with that DB and don't want to play around on the production system and mess something up. So I was hoping for some sort of migration tool, script or at least detailed instructions that I could take and use it to migrate Photoprism.

In the absence of that, it could be that Photoprism's face detection mechanism is much better compared to Synology Photos and that there would be less work than I imagine.
So my follow-up here is: would it be dangerous for me to run Photoprism in parallel on the same photo files alongside the Synology Photos app?

**Reply by graciousgrey** [reply]
> As far as I'm aware (I could be wrong) Synology Photos (as well as previous applications, like Photo Station and Moments) don't touch the files directly, but rather store data in the Synology internal Postgres DB.

If they don't provide an export  we can't read the data.

> So my follow-up here is: would it be dangerous for me to run Photoprism in parallel on the same photo files alongside the Synology Photos app?

This should work. You can set the folder where your original photos are located as the originals directory of PhotoPrism. PhotoPrism does not write to this directory.

**Reply by Vlayke** [reply]
While we were talking to each other I've done some additional research on Photoprism deployment and I've got another migration worry now :-)
On current Synology setup each user has their own location where photos are stored. Each in their own home location. Then there's also a common shared location where (for example) common family photos are located.
So now additional challenges are:

- Can I specify multiple "originals" location for Photoprism so that it would parse all the home locations?
- If it's possible to provide multiple "originals" locations I would it would then completely bypass user privacy and expose all private photos in a common space. That's a no-go.
- If first paragraph is not possible then I could theoretically deploy separate Photoprism instances for each user but that just begs for problems and isn't really a simple common replacement I've been looking for.

So if and when I'd like to actually do a migration this would not be a simple project, but quite an involved one, it appears.
I'm not saying anything is wrong with Photoprism, just that getting into it now looks much harder than I initially thought.

**Reply by graciousgrey** [reply]
You can specify multiple subfolders in originals (https://docs.photoprism.app/getting-started/docker-compose/#photoprismoriginals) but there is only one admin user that has access to all subfolders.

We are currently working on multi user support see: https://github.com/photoprism/photoprism/issues/98.

Before this is finished you would need to set up one instance per user.

**Comment by glenstewart** [comment]
I can add several useful observations to this discussion:

1. The Synology database **does** have entries for the photographs managed by Synology Photos.
2. The photo files EXIF data is not updated nor the file modified as part of the facial recognition process
3. When the disk containing my 62k photos crashed last week, I attempted to use HyperBackup to restore everything.  When the restore completed, all the facial recognition was lost.  All I get are the unrecognized photos I started with.  The sad thing is that the backup did contain the database with recognition data, but it could somehow not RESTORE that, associated to the photos.

In my opinion, users should not accept a non-standardized facial recognition solution that doesn't store the recognition data in the photos themselves, like EXIF.  With PhotoPrism, LibrePhotos, and others in the open source facial recognition effort, creating an open standard for storage in the photo file should be achievable.

**Reply by lastzero** [reply]
There's an XMP based standard we will look into once we have a more powerful XMP / XML parser as we don't want to ship a half baked solution. With PhotoPrism, face recognition data should be restoreable from database backups  - unless the SHA1 hash of all images changes, so the exact files can not be matched anymore.
